### ISSUE TITLE
Compiler dependency tracking introduces non-hermeticity

### ISSUE THREAD

**Issue by thirtyseven** [bazelbuild/rules_scala#1492]
The [http_archive](https://github.com/liucijus/rules_scala/blob/f766a99932e4afaae1cdab81381ab5f101708847/scala/private/macros/scala_repositories.bzl#L25) used to patch scala-compiler does not specify a sha256 hash, resulting in warnings during builds:
```
DEBUG: Rule 'scala_compiler_source' indicated that a canonical reproducible form can be obtained by modifying arguments sha256 = "d57797fe3982d69d56d432046459f5b72e87a422170d98cf295c3b1bbe93f456"
```

Not sure what the best solution here is since rules_scala seems to be moving towards encouraging users to provide their own dependencies via `rules_jvm_external`. Maybe we can write a custom repo rule that takes a label pointing to the srcjar and uses `rtcx.extract` and `rtcx.patch` to patch it.

**Comment by liucijus** [comment]
There are a few ways to address this problem:
1. Override `scala_compiler_source` http_archive, with your own
2. Pass your own label. like in #1493 or equivalent way
3. Hardcode checksums into the macro

1 and 2 are very unfriendly to the user, but 3 is hard to keep up to date. I think we can add #1493, if option 1 is not enough.

**Comment by liucijus** [comment]
closed via #1493
